From: =?utf-8?q?Florian_M=C3=BCllner?= <fmuellner@gnome.org>
Date: Fri, 17 Mar 2023 01:05:48 +0100
Subject: main: Fix syncing animations-enabled

Whether or not animations should be enabled depends on various
factors, some of which may change at runtime. We therefore
track changes, and sync the setting by calling inhibit/uninhibit
as necessary.

Except that we never actually record the new state, so when animations
are disabled, we end up inhibiting them every time
the setting is synced, whoops.

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2698>

Origin: https://gitlab.gnome.org/GNOME/gnome-shell/-/commit/2f196f4
---
 js/ui/main.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/js/ui/main.js b/js/ui/main.js
index 16c1eef..cf9357f 100644
--- a/js/ui/main.js
+++ b/js/ui/main.js
@@ -963,6 +963,7 @@ var AnimationsSettings = class {
         const shouldEnableAnimations = this._shouldEnableAnimations();
         if (this._animationsEnabled === shouldEnableAnimations)
             return;
+        this._animationsEnabled = shouldEnableAnimations;
 
         const settings = St.Settings.get();
         if (shouldEnableAnimations)
